### PR TITLE
Don't require internet access for the runtime to start

### PR DESCRIPTION
**Pull request by mitchdevenport**
This PR makes the necessary changes to allow the runtime to start without internet access. It also makes the necessary change to allow the runtime to start even if a bad dataset is loaded. To do this, on dataset load failure the load is reattempted a configurable number of times. If it is unsuccessful after the configured retries then it gives up.

### CODE REVIEW

File: bin/spiced/src/lib.rs [hunk 1]
```diff
@@ -68,24 +70,14 @@ pub struct Args {
...
-
+    let app = Arc::new(App::new(current_dir.clone()).context(UnableToConstructSpiceAppSnafu)?);
+    let auth = Arc::new(load_auth_providers());
+    let df = Arc::new(RwLock::new(runtime::datafusion::DataFusion::new()));
     let model_map = load_models(&app, &auth);
-
     let pods_watcher = PodsWatcher::new(current_dir.clone());
 
-    let mut rt: Runtime = Runtime::new(args.runtime, app, df, model_map, pods_watcher);
+    let mut rt: Runtime = Runtime::new(args.runtime, app.clone(), df, model_map, pods_watcher);
```
Comment (phillipleblanc) [inline bin/spiced/src/lib.rs]:
```suggestion
    let mut rt: Runtime = Runtime::new(args.runtime, Arc::clone(&app), df, model_map, pods_watcher);
```

File: crates/runtime/src/lib.rs [hunk 2]
```diff
@@ -93,49 +95,81 @@ impl Runtime {
...
+        for ds in self.app.datasets.clone() {
+            self.load_dataset(&ds, auth);
+        }
+    }
 
-        tracing::info!("Loaded dataset: {}", &ds.name);
+    pub fn load_dataset(&self, ds: &Dataset, auth: &Arc<auth::AuthProviders>) {
+        let ds = ds.clone();
+        let df = self.df.clone();
+        let auth = auth.clone();
```
Comment (phillipleblanc) [inline crates/runtime/src/lib.rs]: One reason I prefer using Arc::clone vs .clone() is that for example here - which ones are just cloning a pointer vs cloning the actual object?

Comment (mitchdevenport) [reply crates/runtime/src/lib.rs]: Ugh I know but it feels so much nice to just call .clone!

Will update throughout the PR :)

**Review by lukekim — APPROVED** [review]

**Review by phillipleblanc — APPROVED** [review]
I think we want the Kubernetes model of not giving up on trying to load a dataset - but this is a good first step.

**Pull request merged**